src/ArticleGallery.js

```diff
@@ -0,0 +1,101 @@
+import image from "./assets/images.jpg";
+import firebase from "./firebase.js";
+import {getDatabase, push, ref} from 'firebase/database';
+import {useEffect, useState} from 'react';
+
+
+const ArticleGallery = (props)=>{
+
+    // State for tracking the articles in the user's read list
+    const  [saved, setSaved]=useState({});
+
+
+    // Function to handle the click event and retrieve the properties for image, url and title
+    const handleClick=(e)=>{
+        alert("Article Added");
+
+        // Image Source
+        const source =e.target.parentElement.previousSibling.firstChild.innerHTML;
+
+        //Slice image source to isolate the url
+        const newSource = source.slice(
+        source.indexOf('=') + 2,
+        source.lastIndexOf('alt')- 2,);
+
+         // url property  
+        const link= e.target.parentElement.previousSibling.firstChild.href;
```

Comment:
Dear Joe, 
Nothing like the Simpsons Moe,
You are kind to lead me through this crazy code like Starwars Poe.
This code is flawless and made me think about life and my growth as a man in a pizza pan.
And did you know there is there is pizza place in New York called Joe's?
You should teach me how to react, because my react skills are inapt.
And this APP? 
Is flawless I say. Just the buttons dont add articles as I read about the day.
Be proud and stand tall, 
As the code wind blows over the tennis ball.
You will find an amazing job because you are better than your neighbour Bob.